src/Symfony/Component/Asset/VersionStrategy/JsonManifestVersionStrategy.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\VersionStrategy;
+
+/**
+ * Reads the versioned path of an asset from a JSON manifest file.
+ *
+ * For example, the manifest file might look like this:
+ *     {
+ *         "main.js": "main.abc123.js",
+ *         "css/styles.css": "css/styles.555abc.css"
+ *     }
+ *
+ * You could then as for the version of "main.js" or "css/styles.css".
+ */
+class JsonManifestVersionStrategy implements VersionStrategyInterface
+{
+    private $manifestPath;
+
+    private $manifestData;
+
+    public function __construct($manifestPath)
```

Comment:
please add the phpdoc describing the argument as a string, as we don't have a typehint to tell it